aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -47,9 +47,15 @@ fn get_resource_on_chain_at_addr<T: MoveStructType + for<'a> Deserialize<'a>>(
     module_storage: &impl ModuleStorage,
 ) -> anyhow::Result<T, VMStatus> {
     let struct_tag = T::struct_tag();
+    debug_assert!(struct_tag.address.is_special());
```

Comment:
ditto. this is in keyless, but the API seems too general in a way the method name and address parameter.

Reply:
Yes, here I added a debug assert because of the same thinking. In general, it should always be core code address (special) - maybe just doing a check and returning invariant violation is better? It's not like it's called often and adds much overhead?